pytorch_lightning/core/lightning.py

```diff
@@ -881,6 +881,9 @@ def validation_epoch_end(self, outputs: EPOCH_OUTPUT) -> None:
         Return:
             None
 
+        .. warning:: If you are using `ddp*` accelerators, each process will only see its own data.
```

Comment:
Why a warning? There is nothing misbehaving that needs to be warned about. 
In DDP the data is always partitioned across the devices, that's why it's called data parallel. And it doesn't matter if training, validation or test. I would rather modify the args description, i.e., change

> outputs: List of outputs you defined in ...

to

> outputs: List of outputs gathered from :meth:`validation_step` in the current process, or ...

Reply:
Changed to Note

Reply:
+1 to updating the args description instead. Note/warning blocks are really distracting when reading the docs IMO